cosmos/airflow/graph.py

```diff
@@ -167,32 +168,59 @@ def build_airflow_graph(
     # If test_behaviour=="after_each", each model task will be bundled with a test task, using TaskGroup
     for node_id, node in nodes.items():
         task_meta = create_task_metadata(node=node, execution_mode=execution_mode, args=task_args)
-        if emit_datasets:
-            task_args["outlets"] = [get_dbt_dataset(conn_id, dbt_project_name, node.name)]
+
         if task_meta and node.resource_type != DbtResourceType.TEST:
+            task_outlet = [get_dbt_dataset(connection_id=conn_id, project_name=dbt_project_name, model_name=node.name)]
             if node.resource_type == DbtResourceType.MODEL and test_behavior == TestBehavior.AFTER_EACH:
                 with TaskGroup(dag=dag, group_id=node.name, parent_group=task_group) as model_task_group:
-                    task = create_airflow_task(task_meta, dag, task_group=model_task_group)
-                    test_meta = create_test_task_metadata(
-                        f"{node.name}_test",
-                        execution_mode,
-                        task_args=task_args,
-                        model_name=node.name,
-                        on_warning_callback=on_warning_callback,
-                    )
-                    test_task = create_airflow_task(test_meta, dag, task_group=model_task_group)
-                    task >> test_task
-                    task_or_group = model_task_group
+                    model_task_args = task_args.copy()
+                    if emit_datasets_from == EmitDatasetsType.MODELS:
+                        model_task_args["outlets"] = task_outlet
+
+                    model_meta = create_task_metadata(node=node, execution_mode=execution_mode, args=model_task_args)
+                    if model_meta:
+                        run_model_task = create_airflow_task(model_meta, dag, task_group=model_task_group)
+
+                        test_task_args = task_args.copy()
+                        if emit_datasets_from == EmitDatasetsType.TESTS:
+                            test_task_args["outlets"] = task_outlet
```

Comment:
What happens in this scenario if one of the dbt models doesn't have any tests associated with it?